### PR TITLE
Added PixelColor typedef to make it easier to migrate to ImageMagick 7.

### PR DESCRIPTION
**Pull request by dlemstra**

### CODE REVIEW

**Comment by Watson1978** [comment]
@dlemstra Can you solve the conflicting?

File: ext/RMagick/rmagick.h [hunk 1]
```diff
@@ -1153,6 +1154,7 @@ extern VALUE  rm_enum_new(VALUE, VALUE, VALUE);
 // rmstruct.c
 extern VALUE  ChromaticityInfo_to_s(VALUE);
 extern VALUE  ChromaticityInfo_new(ChromaticityInfo *);
+extern void   Color_to_PixelColor(PixelColor *, VALUE);
 extern void   Color_to_PixelPacket(PixelPacket *, VALUE);
```
Comment (Watson1978) [inline ext/RMagick/rmagick.h]: I think `Color_to_PixelPacket()` should be removed. How do you think?

Comment (dlemstra) [reply ext/RMagick/rmagick.h]: We will still need this inside `Image_pixel_color` where the pixels of an image are accessed. We probably don't need this for ImageMagick7 but we need it for this particular case for now.

**Comment by Watson1978** [comment]
Should we use `PixelColor` for `row_pixels` declaration?
https://github.com/rmagick/rmagick/blob/2c3f41b272ed280a9a79614bb4f38d0e58e95036/ext/RMagick/rmfill.c#L135-L137

**Comment by Watson1978** [comment]
If we should use `PixelColor` instead of `PixelPacket` all, you might find error if you add following `#define` :)

```
typedef PixelPacket PixelColor;
#define PixelPacket #error
```

**Comment by dlemstra** [comment]
That should remain `PixelPacket` because individual pixels will be accessed with `PixelPacket` in ImageMagick 6. For ImageMagick 7 this will be changed into a `Quantum`. The places that will use `PixelInfo` in the future have been renamed to `PixelColor` so we can use an alias.

**Comment by Watson1978** [comment]
ah I understand.

**Comment by dlemstra** [comment]
I am sorry for not making that more clear in the start of the PR. Thanks for your questions so we can get this cleared up.

**Comment by Watson1978** [comment]
I think we should use `PixelColor` in here, right?

https://github.com/rmagick/rmagick/blob/2c3f41b272ed280a9a79614bb4f38d0e58e95036/ext/RMagick/rmfill.c#L447

**Comment by dlemstra** [comment]
Great catch @Watson1978. Pushed a fix for this.

**Comment by Watson1978** [comment]
@dlemstra Can you write a detail of PR in description as possible ? 
When squash and merge the PR, we remain the detail as commit message as possible.

**Pull request merged**